super-ddf.c

```diff
@@ -877,30 +877,151 @@ static void *load_section(int fd, struct ddf_super *super, void *buf,
 	return buf;
 }
 
-static int load_ddf_headers(int fd, struct ddf_super *super, char *devname)
-{
+
+// Search for DDF_HEADER_MAGIC in the last 32MB of the device
+static int search_for_ddf_headers(int fd, char *devname, unsigned long long *out) {
 	unsigned long long dsize;
+	int result = 0;
 
+	// Get device size
 	get_dev_size(fd, NULL, &dsize);
 
-	if (lseek64(fd, dsize - 512, 0) == -1L) {
-		if (devname)
-			pr_err("Cannot seek to anchor block on %s: %s\n",
-			       devname, strerror(errno));
-		return 1;
+	const size_t READ_BLOCK_SIZE = 4096;
+	const unsigned long long SEARCH_REGION_SIZE = 32 * 1024 * 1024;
+
+	// Determine the search range
```

Comment:
For comments, you should use one of:
```
/*Body*/

/*
 * Body
 */
```

Reply:
@slayercat 
Please check https://github.com/md-raid-utilities/mdadm/pull/136/checks